### PR TITLE
More drill work

### PR DESCRIPTION
**Pull request by temp1011**
This is still WIP. At the moment this fixes a bug where using the drill on things like tall grass would still trigger AOE. It seems from what I have tested that `canHarvest()` is broken but I'm not 100%. Still need to stop the drill breaking other stuff when mining dirt etc and get it to work on obsidian etc. I might rewrite more but I'm not really sure at the moment.

### CODE REVIEW

File: src/main/java/techreborn/items/tools/ItemAdvancedDrill.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 package techreborn.items.tools;
 
+import codechicken.lib.raytracer.RayTracer;
```
Comment (modmuss50) [inline src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:27]: You cannot use classes from other mods.

File: src/main/java/techreborn/items/tools/ItemAdvancedDrill.java [hunk 2]
```diff
@@ -76,30 +77,42 @@ public boolean canHarvestBlock(IBlockState blockIn) {
...
 
+	public boolean isGoodAOEBlock(IBlockState state) {
+		return !state.getMaterial().isToolNotRequired() || state.getBlock().isToolEffective("pickaxe",state) || state.getBlock().isToolEffective("shovel",state);
+	}
+
 	@Override
 	public boolean onBlockDestroyed(ItemStack stack, World worldIn, IBlockState blockIn, BlockPos pos, EntityLivingBase entityLiving) {
 		if(!(entityLiving instanceof EntityPlayer))
 			return false;
-		RayTraceResult raytrace= rayTrace(worldIn,  (EntityPlayer) entityLiving, false);
```
Comment (modmuss50) [inline src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:83]: I assume my fix didnt work (I didnt test it as I only had my laptop)

**Review by modmuss50 — CHANGES_REQUESTED** [review]
Dont use classes from other mods.

File: src/main/java/techreborn/items/tools/ItemAdvancedDrill.java (hunk 1 above)
Comment (covers1624) [reply src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:27]: Feel free to copy a stripped down version of `RayTracer` from CCL as done in CoFH Core.

Comment (temp1011) [reply src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:27]: Thanks, I think I will do this.

Comment (modmuss50) [reply src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:27]: What was wrong with my fix? I just tested it and it seemed to work ok?

Comment (temp1011) [reply src/main/java/techreborn/items/tools/ItemAdvancedDrill.java:27]: Nothing, this is a bad branch I am commiting from. Let me reset. I got confused with the workspace set up.

**Comment by temp1011** [comment]
Bad branch.

**Pull request closed without merging**